frontend/controller/actions/group.js

```diff
@@ -130,7 +159,66 @@ export default (sbp('sbp/selectors/register', {
   },
   'gi.actions/group/switch': function (groupId) {
     sbp('state/vuex/commit', 'setCurrentGroupId', groupId)
+    sbp('state/vuex/commit', 'setCurrentChatRoomId', { groupId })
+  },
+  'gi.actions/group/addChatRoom': async function (params: GIActionParams) {
+    const message = await sbp('gi.actions/chatroom/create', { data: params.data })
+
+    const { name, type, privacyLevel } = params.data.attributes
+    await sbp('chelonia/out/actionEncrypted', {
+      action: 'gi.contracts/group/addChatRoom',
+      contractID: params.contractID,
+      data: { name, type, privacyLevel, chatRoomID: message.contractID() }
+    })
```

Comment:
Continuing our discussion about the use of `...params`, I'm noticing that your changes have now made it so that the hooks do not get called in certain situations.

For example, hooks will not get called if:

- `'gi.actions/group/addChatRoom'` is called directly
- `'gi.actions/group/addAndJoinChatRoom'` is called

They will get called if:

- `'gi.actions/group/joinChatRoom'` is called directly (because it does `...params`)

The ideal case is: regardless of which one of these selectors is called, if hooks are passed in, they will be called exactly once only

Reply:
@Silver-IT 

> The ideal case is: regardless of which one of these selectors is called, if hooks are passed in, they will be called exactly once only

I had a thought today on how this can be done fairly easily:

In each of those selectors you do the `...params` thing, but in an action that does that, and calls another action that also does that (e.g. `'gi.actions/group/addAndJoinChatRoom'`), before calling the second action, remove all of the hooks from `params` before calling the action so that they don't get called twice.

Reply:
Yeap. Good point. There is really a mistake here.

Reply:
> In each of those selectors you do the ...params thing, but in an action that does that, and calls another action that also does that (e.g. 'gi.actions/group/addAndJoinChatRoom'), before calling the second action, remove all of the hooks from params before calling the action so that they don't get called twice.

I think if there are several actions calls, we need to send only `hooks.prepublish` to the first action and only `hooks.postpublish` to the last action.
So for the example of `gi.actions/group/addAndJoinChatRoom`, we need to send only `hooks.prepublish` to the `gi.actions/group/addChatRoom` and only `hooks.postpublish` to the `gi.actions/group/joinChatRoom`.

Just want to know if you think so.

Reply:
Sure, @Silver-IT, feel free to do it that way.